Convert QU to Qphi and Uphi

TsutsumiNagai opened the pull request:
## Describe your changes
Added a new function `convert_Q_U_to_Qphi_Uphi()` in `corgidrp/l4_to_tda.py`.

This function:
- Computes the azimuthal angle φ relative to the image center.
- Converts Cartesian Stokes Q and U into azimuthal components Q_phi and U_phi.
- Preserves input cubes (I, Q, U, V) unchanged, and returns modified Qphi and Uphi cubes.
- Includes docstrings and inline comments for clarity.
<img width="926" height="567" alt="image" src="https://github.com/user-attachments/assets/0708eca2-1701-45f0-8baa-88b43bb20756" />



## Type of change
- New feature (non-breaking change which adds functionality)

## Reference any relevant issues (don't forget the #)
[#519](https://github.com/roman-corgi/corgidrp/issues/519)

## Checklist before requesting a review
- [x] I have linted my code
- [x] I have verified that all unit tests pass in a clean environment and added new unit tests, as needed
- [x] I have verified that all docstrings are properly formatted and added new documentation, as needed
- [ ] I have filled out the Unit Test Definition Table on confluence, as needed

maxwellmb reviewed (changes requested):
A few more comments: 
- I misspoke when I said the input should be a `Dataset`. I think it should actually be an `Image` object whose data is a 3D array. My apologies for leading you in the wrong direction. In that case the image will have the shape [4,n,m] where n and m are the sizes in pixels of each "slice", and the four slices are [I,Q,U,V]. In that case you probably don't need to create the function that you used, but instead make the output shape [6,n,m].
- It would keep things as flexible as possible to have the x_center and y_center be allowed keywords to the function that overwrite the default behavior (however, the default should still be using the header keywords like you have already). 
- Along with the main function, please also submit a unit test that demonstrates that this works as expected. Please see the README for some guidance and take a look at the other unit tests in the `tests/` directory. The tests should all be analytical and should be carried out with `assert` statements. You could, for example, test that U_phi is all zero when you get the center correct and non-zero when the center is wrong.

TsutsumiNagai commented:
I have updated the PR with the following changes.
I added a new unit test file, tests/test_compute_QphiUphi.py.
I implemented the calculation of err within compute_QphiUphi.
For the dq handling, I wasn’t sure of the best handling, so currently I just inherit the flags from Q and U.

maxwellmb commented on corgidrp/l4_to_tda.py:
```diff
@@ -541,4 +541,250 @@ def calculate_zero_point(image, star_name, encircled_radius, phot_kwargs=None):
+#     processed_dataset.update_after_processing_step(
+#         history_entry="Appended Q_phi and U_phi."
+#     )
+#     return processed_dataset
```
Please remove these commented lines.

maxwellmb commented:
@TsutsumiNagai, I just noticed there are a few other print statements. If you could remove them please when you finish the other PR responses that would be much appreciated.
